### PR TITLE
Feature setup

### PR DESCRIPTION
**Pull request by simandebvu**
We created a secret post page that allows us to build posts that belong to different users and we establish numerous methods of validations so the information provided will be one acceptable, and those who are not signed can see the post, but not the name of the one who made it

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
## Hi team :clap: 

### You did a great job so far! :+1:  , just a few changes before it can be approved:


#### Users

##### You should implement Users Controller

##### `new` action

* [ ] it has a `new` action with `User.new`  from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/)
screenshot

##### `create` action

* [ ] it has a create action. Must use `user_params`  from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/)
screenshot 

##### view

* [ ] It has a sign up form created with #form_for in `app/views/users` from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/)
screenshot 

#### Posts

##### routes

* [ ] It creates the application routes using `resources :posts, only: [:new, :create, :index]` from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)
screenshot 


Kindly make the changes and submit a new code review request.
Thanks!

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by simandebvu** [comment]
Changes have been made - however as this was a devise based project, the required actions and controllers are in the devise `registrations-controller`.

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <td><%= post.user.email %>
+        <% end %>
+        <td><%= post.body %></td>
+        <td><%= link_to 'Show', post %></td>
+        <% if user_signed_in? && current_user.id == post.user_id %>
+        <td><%= link_to 'Edit', edit_post_path(post) %></td>
+        <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% end %>
+      </tr>
+    <% end %>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb:25]: This is what we call **logic in views**.
Use **view helpers** to implement it.

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<p id="notice"><%= notice %></p>
+
+<p>
+  <strong>Body:</strong>
+  <%= @post.body %>
+</p>
+<% if user_signed_in? && current_user.id == @post.user_id %>
+<%= link_to 'Edit', edit_post_path(@post) %> |
+<% end %>
```
Comment (maelfosso) [inline app/views/posts/show.html.erb:9]: Use **view helpers** for logic, please

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Rails.application.routes.draw do
+  devise_for :users, :controllers=> {registrations:'registrations'}
+  resources :posts, only: [:new, :create, :index, :show]
```
Comment (maelfosso) [inline config/routes.rb:3]: As indicated by the last reviewer and [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png), the routes are restricted.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:2]: Some of those methods are not allowed. 
Kindly check the allowed methods [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png)

Comment (simandebvu) [reply app/controllers/posts_controller.rb:2]: Routes were updated

File: config/routes.rb (hunk 3 above)
Comment (simandebvu) [reply config/routes.rb]: Routes were updated

**Pull request merged**